src/utils/warnings.ts

```diff
@@ -5,23 +5,60 @@
 
 /** A {@link WarningHandler} is called when a new warning is emitted */
 export type WarningHandler = (message: string) => void;
+export type WarningSource = object | null;
+
+/** A minimalistic class to handle warnings */
+export class Warnings {
+    private handlersList: WarningHandler[] = [
+        // eslint-disable-next-line no-console
+        (message) => console.warn(message),
+    ];
+
+    /** Adds a warning handler to the list */
+    public add(handler: WarningHandler) : void {
+        this.handlersList.push(handler);
+    }
+
+    /** Sends a warning message though all handlers */
+    public send(message: string) : void{
+        for (const cb of this.handlersList) {
+            console.log(this.handlersList.length, ': using handler ', cb)
+            cb(message);
+        }
+    }
+}
```

Comment:
We could keep the name `Warning` for this class if `send` is a member function; but then I'd rename `add` to `addHandler`, and maybe add a `clearHandlers` function?

Reply:
for the moment I just renamed add. Now we've one per visualizer I don't see a use case to remove them